### PR TITLE
Add OIDC upload

### PR DESCRIPTION
**Pull request by freundTech**

### CODE REVIEW

**Comment by freundTech** [comment]
CC: @wolfv, @pavelzw

**Comment by freundTech** [comment]
I needed to add some more quotes to the provided yaml to make shellcheck happy.
Because I only upload on main I had to merge this already to see if it works.

The yaml is a bit long and hard to read. I guess this could be turned into an action in the future?

**Comment by wolfv** [comment]
yes, definitely the goal is to hide this complexity e.g. in the rattler-build action! :) Thank you so much for testing!

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -34,11 +34,29 @@ jobs:
       with:
         path: |
           output/**/*.conda
+
+    - name: Get JWT
+      id: get_token
+      if: github.event_name == 'push'  # do not upload on PR
+      uses: actions/github-script@v6
```
Comment (pavelzw) [inline .github/workflows/build.yml:41]: i would suggest pinning this by hash and adding a comment for the version for stability as in https://github.com/freundTech/typst-forge/blob/13ee793295365281bdb95c858a68dc3eca0b7b94/.github/workflows/pre-commit.yml#L16

dependabot can update the comment

**Pull request merged**